refactor(coral): Update modal with topic request details for CLAIM requests

programmiri opened the pull request:
# Linked issue

Resolves: #1679 

# What kind of change does this PR introduce?

- [ ] Bug fix
- [ ] New feature
- [x] Refactor
- [ ] Docs update
- [ ] CI update

# What is the current behavior?

The details modal show values for topic partition, topic description and topic replication factor for `CLAIM` requests, but the endpoint does not return them. 


#### Update request

<img width="900" alt="Screenshot 2023-12-27 at 16 19 48" src="https://github.com/Aiven-Open/klaw/assets/943800/e306dc4d-d851-42ef-8ff8-b9705aeed005">


#### CLAIM request (before change)

<img width="945" alt="Screenshot 2023-12-27 at 16 20 46" src="https://github.com/Aiven-Open/klaw/assets/943800/2a8ff516-febc-4f06-b6b3-9677eedebbff">



# What is the new behavior?

If it's a claim request, the headlines are not shown.

<img width="940" alt="Screenshot 2023-12-27 at 16 20 29" src="https://github.com/Aiven-Open/klaw/assets/943800/af242f9b-3f3f-4976-8c33-c196b646132a">


# Requirements (all must be checked before review)

- [x] The pull request title follows [our guidelines](https://github.com/Aiven-Open/klaw/blob/main/CONTRIBUTING.md#guideline-commit-messages)
- [x] Tests for the changes have been added (if relevant)
- [x] The latest changes from the `main` branch have been pulled
- [x] `pnpm lint` has been run successfully

programmiri commented on coral/src/app/features/components/TopicDetailsModalContent.tsx:
```diff
@@ -73,20 +73,30 @@ const TopicDetailsModalContent = ({
-        <Label>Topic replication factor</Label>
-        <dd>{replicationfactor}</dd>
-      </Box.Flex>
+      {/*getTopicRequestsForApprover for a CLAIM request*/}
```
I hope the comments helps future 👀  seeing this code and explain why it's dependent on "CLAIM" request. This could also be a backend fix in the future to make openapi more safe.

mathieu-anderson reviewed (commented):
Oh, thank you! Should we also render requestor's team in this context? We don't do it for other requests, but I think it is very relevant information for CLAIM, as ownership will pass to them :o

programmiri commented:
>Should we also render requestor's team in this context? 

Yes, that sounds like a good idea! We have it in the table, but I think it's good to have it in the modal, too! 

Will update!

programmiri commented:
Updated! https://github.com/Aiven-Open/klaw/pull/2139/commits/d5ee658703935d66e69384797f5b1afc3d9d2f97

I added it besides "Requested by" because I thought the information made the most sense there?

<img width="967" alt="Screenshot 2024-01-08 at 11 42 23" src="https://github.com/Aiven-Open/klaw/assets/943800/0c7326b2-c73c-4974-9992-5c46368e118f">
